src/mlpack/tests/init_rules_test.cpp

```diff
@@ -126,6 +129,35 @@ BOOST_AUTO_TEST_CASE(OivsInitTest)
   BOOST_REQUIRE_EQUAL(1, 1);
 }
 
+BOOST_AUTO_TEST_CASE(XavierInitTest)
+{
+  arma::mat weights;
+  //RandomSeed(21);
+  XavierInit xavierUniform;
+  XavierInit xavierNormal;
+  xavierUniform.Initialize<XavierUniform>(weights, 100, 100);
```

Comment:
Since the ``Initialize`` function is used inside e.g. the FFN class there is no easy way to call  `` xavierUniform.Initialize<XavierUniform>`` since, there is no option for a user to provide XavierUniform. An easy solution would be to templatize the ``XavierInit`` class. What do you think?